javascript/ql/src/Security/CWE-116/UnsafeHtmlExpansion.qhelp

```diff
@@ -0,0 +1,100 @@
+<!DOCTYPE qhelp PUBLIC
+"-//Semmle//qhelp//EN"
+"qhelp.dtd">
+<qhelp>
+
+	<overview>
+		<p>
+
+			Sanitizing untrusted input for HTML meta-characters is an
+			important technique for preventing cross-site scripting attacks. But
+			even a sanitized input can be dangerous to use if it is modified
+			further before a browser treats it as HTML.
+
+			A seemingly innocent transformation that expands a
+			self-closing HTML tag from <code>&lt;div attr="{sanitized}"/&gt;</code>
+			to <code>&lt;div attr="{sanitized}"&gt;&lt;/div&gt;</code> may
+			in fact cause cross-site scripting vulnerabilities.
+
+		</p>
+		
+	</overview>
+
+	<recommendation>
+		<p>
+
+			Use a (well-tested) sanitization library if at all
+			possible, and avoid modifying sanitized values further before treating
+			them as HTML.
+
+		</p>
+	</recommendation>
+
+	<example>
+
+		<p>
+
+			The following function transforms a self-closing HTML tag
+			to a pair of open/close tags.  It does so for all non-<code>img</code>
+			and non-<code>area</code> tags using a regular expression with two
+			capture groups. The first capture group corresponds to the name of the
+			tag, and the second capture group corresponds to the content of
+			the tag.
+
+		</p>
+
+		<sample src="examples/UnsafeHtmlExpansion.js" />
+
+		<p>
+
+			While it is generally known regular expressions are
+			ill-suited for parsing HTML, variants of this particular transformation
+			pattern have long been considered safe.
+
+		</p>
+
+		<p>
+
+			However, the function is not safe. As an example, consider
+			the following string which does not result in an alert when a
+			browser treats it as HTML:
```

Comment:
Suggesting the following, which is shorter, and avoids the repetition with the next sentence:
However, the function is not safe. As an example, consider the following string: